### PR TITLE
Update Navigation with vanilla framework 3.7.1

### PR DESCRIPTION
**Pull request by edlerd**
Using the latest and greatest Vanilla native navigation. It is a little different from current https://linuxcontainers.org/ mainly dropped the outlines and made it easier to read.

Further improvements with new vanilla effect boldness of menu, h2 and h3 text. Along with spacing refinements.

Do you agree and are you fine with the new styles (see images below)?

There are some open issues, which would require more work. I pause them because first want to know if the new styles are desired.

If the answer is yes, I'd invest more time to fix the remaining issues to make this mergeable. If not, the site could just stay with the old vanilla version, navigation, etc.

New menu closed

new-vanilla-menu-closed

New menu open

new-vanilla-menu-open

Signed-off-by: David Edler <david.edler@canonical.com>

### CODE REVIEW

**Comment by tomponline** [comment]
I suppose if we do change it, will this also need updating? 

https://linuxcontainers.org/lxd/docs/master/

**Comment by ru-fu** [comment]
Overall, I think it would be good to use the latest framework, but I'm not sure how much effort it's worth spending on it.

For the website, you should check the online demo and see if that one works with the update; it's probably the most complex part of the website.

> I suppose if we do change it, will this also need updating?
> 
> https://linuxcontainers.org/lxd/docs/master/

Yes, the docs should look as similar to the homepage as possible.
They aren't using the Vanilla stylesheet, because it interfered with the Furo theme (if I remember correctly). So they would need to be adapted manually.
The header is here: https://github.com/lxc/lxd/blob/master/.sphinx/_templates/header.html

**Comment by edlerd** [comment]
> I suppose if we do change it, will this also need updating?
> 
> https://linuxcontainers.org/lxd/docs/master/

As far as I knot, the documentation pages are not using vanilla. I think they are not generated from this repo.

We could change them to use vanilla as well, though, in order to align the menus.

**Comment by stgraber** [comment]
In general, I'm happy to have us refresh to current vanilla.
We used to get PRs from the web team every few months or year to do so, but that hasn't happened in a while, so I suspect there's some catch up to do :)

We'll just want @ru-fu to take a close look at the changes prior to merging, but assuming it doesn't break anything (the online demo is the most likely to break), then we're happy to do it.

**Comment by edlerd** [comment]
Fixed all known issues with this PR.

Also created a PR on LXD to make the header in [the docs](https://linuxcontainers.org/lxd/docs/) look like the new one here for linuxcontainers.org.

File: generate
Comment (ru-fu) [inline generate:99]: Why is this not needed anymore? Isn't this what generates the sub-menu?

(I don't have the manpages locally to test.)

File: static/css/local.css
Comment (ru-fu) [inline static/css/local.css:363]: Need to add a line break at the end of the file.

**Review by ru-fu — COMMENTED** [review]
Some findings - I built the page locally, so there might be build issues on my side causing what I see.

Navigation:
- Introduction is missing for all menus?!

Styling:
- Space for the breadcrumbs is bigger and breadcrumbs aren't vertically centered (might be intended though)
- The line under tabs is not aligned (see LXD > Getting started > Other installation options)

Online demo:
- Start button for online demo is placed strangely
- The start button doesn't turn green, but the final button for submitting feedback is green
- Are the step numbers supposed to not have all lines?
- Can we decrease the space/font for the online demo to fit more content on the screen?
- Why is the terminal a different width than the instructions?
- "Previous" button on first page (and "Next" on last) should be disabled
- We shouldn't have "i" icons in the demo, they are confusing (since you can't click on them to display more information)

File: generate
Comment (edlerd) [reply generate:99]: The vanilla menu doesn't support 2nd level menus. I talked to vanilla dev, and he mentioned it is considered bad UX. Hence, I suggest linking to the normal page. Like https://linuxcontainers.org/lxc/manpages//index.html already exists. This is what current PR is doing, and I removed this generation code for 2nd level menus.

**Comment by edlerd** [comment]
Thank you for the review @ru-fu !
All points mentioned should be resolved. Please let me know if anything else pops up that is not as expected.

**Review by ru-fu — APPROVED** [review]
Looks good to me overall! :)

In the online demo, the terminal still has a different width than the instructions. It works though, so this isn't a blocking issue.

For the man pages, I guess the solution will work (as mentioned I can't test this locally). But leaving the decision up to @stgraber if we're okay with getting rid of the submenu.

**Comment by edlerd** [comment]
Adjusted the demo page. Terminal aligns now with the instruction container width.

**Comment by stgraber** [comment]
@ru-fu can you take another quick pass to make sure it all looks good now?

Then we can get the commits rebased/re-titled to look better and we can merge this.

**Comment by ru-fu** [comment]
> @ru-fu can you take another quick pass to make sure it all looks good now?

Yes, looks all good to me. :)

**Comment by edlerd** [comment]
The menu is collapsing to two rows in screen widths between ~600 and ~800 pixels. Vanilla release 3.8.0 will contain a fix. It is a tiny issue though, not critical.
But maybe let's hold off the merge until that one is ready and referenced from this PR?

**Comment by edlerd** [comment]
Updated to Vanilla 3.8.0 and squashed all commits in one. Ready to merge?

**Pull request merged**
